src/erlang_ls_utils.erl

```diff
@@ -12,16 +13,25 @@
 %%
 %% Look for a given module in the DB and return the respective
 %% `URI`. If the module is not in the DB, try to index it.
+
 -spec find_module(atom()) -> {ok, uri()} | {error, any()}.
 find_module(M) ->
-  case erlang_ls_db:find(completion_index, M) of
+  find_module(M, erl).
+
+-spec find_module(atom(), erl | hrl) -> {ok, uri()} | {error, any()}.
+find_module(M, Extension) ->
+  case erlang_ls_db:find(modules, M) of
     {ok, Uri} ->
       {ok, Uri};
     {error, not_found} ->
-      FileName = atom_to_list(M) ++ ".erl",
-      erlang_ls_index:find_and_index_file(FileName)
+      FileName = atom_to_list(M) ++ extension(Extension),
+      erlang_ls_indexer:find_and_index_file(FileName, sync)
```

Comment:
We need this indexing to be synchronous, since we rely on the file having been indexed after this function returns.